test/test_overrides.py

```diff
@@ -587,6 +587,16 @@ def instance_gen():
                     raise RuntimeError(f"Unsupported argument type {t} for {arg['name']} of function {func}")
         else:
             args = inspect.getfullargspec(override)
+            try:
+                func_args = inspect.getfullargspec(func)
+                # Remove annotations from argspec
+                func_args = type(func_args)(**{**func_args, 'annotations': None})
+                if func_args != args:
+                    raise RuntimeError(f"Override for {func} doesn't match its argspec.\n"
+                                       + f"Original: {inspect.signature(func)}\n"
+                                       + f"Override: {inspect.signature(override)}")
+            except TypeError:
+                pass
```

Comment:
nit: Scope the try-except around the exact statement you expect to throw; that will prevent spuriously catching errors. In your case, you want to run some code if you manage to get through it without throwing: you can use an `else:` block after the `except:` to do this.